packages/typespec-test/test/openai_generic/generated/typespec-ts/src/api/chat/completions/index.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  chatCompletionRequestMessageSerializer,
+  chatCompletionFunctionsSerializer,
+  CreateChatCompletionRequest,
+  CreateChatCompletionResponse,
+} from "../../../models/models.js";
+import {
+  ChatCompletionsCreate200Response,
+  ChatCompletionsCreateDefaultResponse,
+  isUnexpected,
+  OpenAIContext as Client,
+} from "../../../rest/index.js";
+import {
+  StreamableMethod,
+  operationOptionsToRequestParameters,
+  createRestError,
+} from "@azure-rest/core-client";
+import { serializeRecord } from "../../../helpers/serializerHelpers.js";
+import { ChatCompletionsCreateOptionalParams } from "../../../models/options.js";
+
+export function _createSend(
+  context: Client,
+  body: CreateChatCompletionRequest,
+  options: ChatCompletionsCreateOptionalParams = { requestOptions: {} },
+): StreamableMethod<
+  ChatCompletionsCreate200Response | ChatCompletionsCreateDefaultResponse
+> {
+  return context
+    .path("/chat/completions")
+    .post({
+      ...operationOptionsToRequestParameters(options),
+      body: {
+        model: body["model"],
+        messages: body["messages"].map(chatCompletionRequestMessageSerializer),
+        functions:
+          body["functions"] === undefined
+            ? body["functions"]
+            : body["functions"].map(chatCompletionFunctionsSerializer),
+        function_call: body["functionCall"],
+        temperature: body["temperature"],
+        top_p: body["topP"],
+        n: body["n"],
+        max_tokens: body["maxTokens"],
+        stop: body["stop"],
+        presence_penalty: body["presencePenalty"],
+        frequency_penalty: body["frequencyPenalty"],
+        logit_bias: !body.logitBias
+          ? body.logitBias
+          : (serializeRecord(body.logitBias as any) as any),
+        user: body["user"],
+        stream: body["stream"],
+      },
+    });
+}
+
+export async function _createDeserialize(
+  result:
+    | ChatCompletionsCreate200Response
+    | ChatCompletionsCreateDefaultResponse,
+): Promise<CreateChatCompletionResponse> {
+  if (isUnexpected(result)) {
+    throw createRestError(result);
+  }
+
+  return {
+    id: result.body["id"],
+    object: result.body["object"],
+    created: new Date(result.body["created"]),
+    model: result.body["model"],
+    choices: result.body["choices"].map((p) => ({
+      index: p["index"],
+      message: {
```

Comment:
Or maybe union,

Reply:
Do you remember the repro of the issues you mention? This is being generated like this because we get an anonymous model, some of the properties here do call a serializer thoguh, which should resolve recursive issues I think